godotopenxrpico/src/main/java/org/godotengine/openxr/vendors/pico/GodotOpenXRPico.kt

```diff
@@ -48,11 +63,116 @@ class GodotOpenXRPico(godot: Godot?) : GodotPlugin(godot) {
+         *
+         * TODO: remove this when Godot 4.3 is out with the new `PermissionsUtil.requestPermissions(...)` api
+         */
+        private fun requestPermissions(activity: Activity?, permissions: List<String>): Boolean {
+            if (activity == null) {
+                return false
+            }
+            if (permissions.isEmpty()) {
+                return true
+            }
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+                // Not necessary, asked on install already
+                return true
+            }
+
+            val requestedPermissions: MutableSet<String> = HashSet()
+            for (permission in permissions) {
+                try {
+                    if (permission == Manifest.permission.MANAGE_EXTERNAL_STORAGE) {
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R && !Environment.isExternalStorageManager()) {
+                            Log.d(TAG, "Requesting permission $permission")
+                            try {
+                                val intent = Intent(Settings.ACTION_MANAGE_APP_ALL_FILES_ACCESS_PERMISSION)
+                                intent.setData(Uri.parse(String.format("package:%s", activity.packageName)))
+                                activity.startActivityForResult(intent, PermissionsUtil.REQUEST_MANAGE_EXTERNAL_STORAGE_REQ_CODE)
+                            } catch (ignored: Exception) {
+                                val intent = Intent(Settings.ACTION_MANAGE_ALL_FILES_ACCESS_PERMISSION)
+                                activity.startActivityForResult(intent, PermissionsUtil.REQUEST_MANAGE_EXTERNAL_STORAGE_REQ_CODE)
+                            }
+                        }
+                    } else {
+                        val permissionInfo = getPermissionInfo(activity, permission)
+                        val protectionLevel = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) permissionInfo.protection else permissionInfo.protectionLevel
+                        if (protectionLevel == PermissionInfo.PROTECTION_DANGEROUS && activity.checkSelfPermission(permission) !== PackageManager.PERMISSION_GRANTED) {
+                            Log.d(TAG, "Requesting permission $permission")
+                            requestedPermissions.add(permission)
+                        }
+                    }
+                } catch (e: PackageManager.NameNotFoundException) {
+                    // Skip this permission and continue.
+                    Log.w(TAG, "Unable to identify permission $permission", e)
+                }
+            }
+            if (requestedPermissions.isEmpty()) {
+                // If list is empty, all of dangerous permissions were granted.
+                return true
+            }
+            activity.requestPermissions(requestedPermissions.toTypedArray<String>(), PermissionsUtil.REQUEST_ALL_PERMISSION_REQ_CODE)
+            return true
+        }
```

Comment:
Once you rebase on top of https://github.com/GodotVR/godot_openxr_vendors/pull/176, you can remove these methods; the updated version of Godot 4.3 we're targeting already includes them.